Formula/camlp-streams.rb

```diff
@@ -0,0 +1,21 @@
+class CamlpStreams < Formula
+  desc "Stream and Genlex libraries for use with Camlp4 and Camlp5"
+  homepage "https://github.com/ocaml/camlp-streams"
+  url "https://github.com/ocaml/camlp-streams/archive/refs/tags/v5.0.1.tar.gz"
+  sha256 "ad71f62406e9bb4e7fb5d4593ede2af6c68f8b0d96f25574446e142c3eb0d9a4"
+  license "LGPL-2.1-only" => { with: "OCaml-LGPL-linking-exception" }
+
+  depends_on "dune" => :build
+  depends_on "ocaml-findlib" => :test
+  depends_on "ocaml"
+
+  def install
+    system "dune", "build", "@install"
+    system "dune", "install", "--prefix=#{prefix}", "--libdir=#{lib}/ocaml", "--docdir=#{doc.parent}"
+  end
+
+  test do
+    output = shell_output("#{Formula["ocaml-findlib"].opt_bin}/ocamlfind query camlp-streams")
+    assert_equal "#{HOMEBREW_PREFIX}/lib/ocaml/camlp-streams", output.chomp
```

Comment:
Not sure how to test this so currently just checking if library is available. If anyone has an example test, would be good to update.

---

On side note, the formula will fail notability audit, but I think we can bypass. Already packaged by Arch, Debian/Ubuntu, Fedora, Gentoo, nixpkgs, and is also a requirement for `camlp5`.

Ref: https://repology.org/project/camlp-streams/versions, https://repology.org/project/ocaml:camlp-streams/versions